src/benchmarks/micro/corefx/System.Buffers/ReadOnlySequenceTests.cs

```diff
@@ -0,0 +1,188 @@
+            const int segmentSize = Size / segmentsCount;
+            _startSegment = new BufferSegment<T>(new ReadOnlyMemory<T>(_array.Take(segmentSize).ToArray()));
+            _endSegment = _startSegment;
+            for (int i = 1; i < segmentsCount; i++)
+            {
+                _endSegment = _endSegment.Append(new ReadOnlyMemory<T>(_array.Skip(i * segmentSize).Take(Size / segmentsCount).ToArray()));
+            }
+        }
+
+        [Benchmark]
+        public int IterateTryGetTenSegments()
+            => IterateTryGet(new ReadOnlySequence<T>(startSegment: _startSegment, startIndex: 0, endSegment: _endSegment, endIndex: Size / 10));
+
+        [Benchmark]
+        public int IterateForEachTenSegments()
+            => IterateForEach(new ReadOnlySequence<T>(startSegment: _startSegment, startIndex: 0, endSegment: _endSegment, endIndex: Size / 10));
+
+        [Benchmark]
+        public int IterateGetPositionTenSegments()
+            => IterateGetPosition(new ReadOnlySequence<T>(startSegment: _startSegment, startIndex: 0, endSegment: _endSegment, endIndex: Size / 10));
+
+        [Benchmark(OperationsPerInvoke = 16)]
+        public int FirstTenSegments()
+            => First(new ReadOnlySequence<T>(startSegment: _startSegment, startIndex: 0, endSegment: _endSegment, endIndex: Size / 10));
+
+        [Benchmark(OperationsPerInvoke = 10)]
+        public long SliceTenSegments()
+            => Slice(new ReadOnlySequence<T>(startSegment: _startSegment, startIndex: 0, endSegment: _endSegment, endIndex: Size / 10));
+
+        [MethodImpl(MethodImplOptions.NoInlining)] // make sure that the method does not get inlined for any of the benchmarks and we compare apples to apples
+        private int IterateTryGet(ReadOnlySequence<T> sequence)
+        {
+            int consume = 0;
+
+            SequencePosition position = sequence.Start;
+            while (sequence.TryGet(ref position, out var memory))
+                consume += memory.Length;
+
+            return consume;
+        }
+
+        [MethodImpl(MethodImplOptions.NoInlining)]
+        private int IterateForEach(ReadOnlySequence<T> sequence)
+        {
+            int consume = 0;
+
+            foreach (ReadOnlyMemory<T> memory in sequence)
+                consume += memory.Length;
+
+            return consume;
+        }
+
+        [MethodImpl(MethodImplOptions.NoInlining)]
+        private int IterateGetPosition(ReadOnlySequence<T> sequence)
+        {
+            int consume = 0;
+
+            SequencePosition position = sequence.Start;
+            int offset = (int)(sequence.Length / 10);
+            SequencePosition end = sequence.GetPosition(0, sequence.End);
+
+            while (!position.Equals(end))
+            {
+                position = sequence.GetPosition(offset, position);
+                consume += position.GetInteger();
+            }
+
+            return consume;
+        }
+
+        [MethodImpl(MethodImplOptions.NoInlining)]
```

Comment:
Why?

Reply:
the first method with this attribute above contains the explanation:

```cs
// make sure that the method does not get inlined for any of the benchmarks and we compare apples to apples
```

should I copy this comment for every use? I personally always add the explanation to the first appearance of it in the file

Reply:
I see.  Ok.